pick.lic

```diff
@@ -2,523 +2,665 @@
+      DRC.message('***ITEMS ARE STILL IN HANDS, EXITING***')
+      @equipment_manager.wear_items(@removed_items)
+      exit
+    end
+  end
 
-      find_lockpick unless @settings.use_lockpick_ring
+  def setup_flags
+    Flags.add('disarm-shift', 'something to shift')
+    Flags.add('disarm-trap-type', *@all_trap_messages.values)
+    Flags.add('more-traps', 'not fully disarmed', 'not yet fully disarmed', 'still has more to torment you with')
+    Flags.add('more-locks', 'You discover another lock protecting')
+    Flags.add('glance-no-traps', 'It looks like there are no traps left on')
+    Flags.add('glance-no-locks', 'It looks like there are no locks left on')
+  end
 
-      loop do
-        if DRSkill.getxp('Locksmithing') >= 30
-          bput("put my #{box} in my #{source}", 'You put your .* in ') if right_hand
-          fput('stow left') unless @settings.use_lockpick_ring
-          break
-        end
+  def crack_boxes
+    @boxes_by_bag.each do |source, boxes|
+      boxes.each do |box_noun|
+        break if stop_picking?
 
-        case bput("pick my #{box} blind", 'not even locked', 'Roundtime', 'Find a more appropriate tool')
-        when 'not even locked'
-          waitrt?
-          fput('stow left') unless @settings.use_lockpick_ring
-          loot(box)
-          dismantle(box)
-          break
-        when 'Find a more appropriate tool'
-          bput("put my #{box} in my #{source}", 'You put your .* in ') if right_hand
-          fput('stow left') unless @settings.use_lockpick_ring
-          break
-        end
+        do_buffs
+        DRC.bput('sit', 'You sit', 'You are already sitting') unless sitting?
+        DRCI.get_item?(box_noun, source)
+        attempt_open(box_noun)
       end
+      break if stop_picking?
     end
 
-    refill_ring
+    DRC.fix_standing
+    @equipment_manager.wear_items(@removed_items)
   end
 
   def stop_picking?
     @stop_pick_on_mindlock && DRSkill.getxp('Locksmithing') >= 30
   end
 
-  def setup
-    arg_definitions = [
-      [
-        { name: 'pets', regex: /pets/, optional: true, description: 'Disarm boxes and place them in the pet box container.' },
-        { name: 'count', regex: /\d+/, optional: true, description: 'How many pet boxes to make.' },
-        { name: 'refill', regex: /refill/, optional: true, description: 'Refills your lockpick ring.' },
-        { name: 'source', regex: /(source=(\w+))/, optional: true, description: 'Container with boxes to pick.'}
-      ]
-    ]
-
-    args = parse_args(arg_definitions, true)
-
-    @make_pets = args.pets
-    @pet_goal = args.count.to_i
-    @pet_count = 0
-
-    @settings = get_settings(args.flex)
-
-    if args.source
-      source = args.source.split('=')[1]
-      @settings.picking_box_source = source
+  def do_buffs
+    if @settings.lockpick_buff_bot && DRRoom.pcs.include?(@settings.lockpick_buff_bot)
+      DRC.bput("whisper #{@settings.lockpick_buff_bot} buff hol", "You whisper")
```

Comment:
What about moving `hol`, the requested buff spell, to a setting?

HOL (Hands of Lirisa) is a Ranger spell, but there's also Drums of the Snake (Bard) and Machinist's Touch (Moon Mage) that someone might request of their buff bot depending on which character they set up.

Reply:
No strong feelings on this. This is holdover from the old version of pick. It's a bigger change to do this without breaking anyone who expects it to just be HoL though.